app/renderer/components/Inspector/Screenshot.js

```diff
@@ -1,70 +1,80 @@
-import B from 'bluebird';
+import { Spin } from 'antd';
 import styles from './Inspector.css';
 import { SCREENSHOT_INTERACTION_MODE, INTERACTION_MODE, POINTER_TYPES,
          DEFAULT_TAP, DEFAULT_SWIPE } from './shared';
 
 const { POINTER_UP, POINTER_DOWN, PAUSE, POINTER_MOVE } = POINTER_TYPES;
-const { TAP, SELECT, SWIPE } = SCREENSHOT_INTERACTION_MODE;
+const { TAP, SELECT, SWIPE, TAP_SWIPE } = SCREENSHOT_INTERACTION_MODE;
 const TYPES = {FILLED: 'filled', NEW_DASHED: 'newDashed', WHOLE: 'whole', DASHED: 'dashed'};
 
 /**
  * Shows screenshot of running application and divs that highlight the elements' bounding boxes
  */
 const Screenshot = (props) => {
-  const { screenshot, mjpegScreenshotUrl, methodCallInProgress, screenshotInteractionMode, swipeStart, swipeEnd,
-          scaleRatio, selectedTick, selectedInteractionMode, applyClientMethod, t } = props;
+  const { screenshot, mjpegScreenshotUrl, methodCallInProgress, screenshotInteractionMode, coordStart, coordEnd,
+          setCoordEnd, setCoordStart, scaleRatio, selectedTick, selectedInteractionMode, applyClientMethod, t, clearCoordAction } = props;
 
   const containerEl = useRef();
   const [x, setX] = useState();
   const [y, setY] = useState();
 
   const handleScreenshotClick = async () => {
-    const { setSwipeStart, setSwipeEnd, tapTickCoordinates } = props;
-    const { POINTER_NAME, DURATION_1, DURATION_2, BUTTON } = DEFAULT_TAP;
+    const { tapTickCoordinates } = props;
+    if (selectedTick) {
+      await tapTickCoordinates(x, y);
+    }
+  };
 
-    if (screenshotInteractionMode === TAP) {
-      applyClientMethod({
-        methodName: TAP,
-        args: [
-          {
-            [POINTER_NAME]: [
-              {type: POINTER_MOVE, duration: DURATION_1, x, y},
-              {type: POINTER_DOWN, button: BUTTON},
-              {type: PAUSE, duration: DURATION_2},
-              {type: POINTER_UP, button: BUTTON}
-            ],
-          }
-        ],
-      });
-    } else if (selectedTick) {
-      tapTickCoordinates(x, y);
-    } else if (screenshotInteractionMode === SWIPE) {
-      if (!swipeStart) {
-        setSwipeStart(x, y);
-      } else if (!swipeEnd) {
-        setSwipeEnd(x, y);
-        await B.delay(500); // Wait a second to do the swipe so user can see the SVG line
-        await handleDoSwipe({x, y}); // Pass swipeEnd because otherwise it is not retrieved
+  const handleScreenshotDown = async () => {
+    if (screenshotInteractionMode === TAP_SWIPE) {
+      await setCoordStart(x, y);
+    }
+  };
+
+  const handleScreenshotUp = async () => {
+
+    if (screenshotInteractionMode === TAP_SWIPE) {
+      await setCoordEnd(x, y);
+      if (Math.abs(coordStart.x - x) < 5 && Math.abs(coordStart.y - y) < 5) {
+        await handleDoTap({x, y}); // Pass coordEnd because otherwise it is not retrieved
+      } else {
+        await handleDoSwipe({x, y}); // Pass coordEnd because otherwise it is not retrieved
       }
     }
```

Comment:
Would suggest putting `clearCoordAction();` here and removing it from `handleDoTap` and `handleDoSwipe`. You can then also retrieve `clearCoordAction` in a destructuring call at the start of this method, and remove it from the top-level `props` destructuring call.

Reply:
Yes, agree on all the comments, fixed